src/quantize.rs

```diff
@@ -256,18 +256,13 @@ impl QuantizationContext {
   {
     // Find the last non-zero coefficient using our smaller biases and
     // zero everything else.
-    let mut is_zero = true;
-    let mut pos = coded_tx_size - 1;
-    while is_zero && pos != 1 {
-      let c = coeffs[pos] << (self.log_tx_scale as usize);
-      let qc = c + (c.signum() * T::cast_from(self.ac_offset_eob));
-      is_zero =
-        T::cast_from(divu_pair(qc.as_(), self.ac_mul_add)) == T::cast_from(0);
-      pos -= 1;
-    }
-
+    let deadzone = ((1 << self.log_tx_scale) - 1 + self.ac_quant as i32
+      - self.ac_offset_eob)
+      >> self.log_tx_scale;
+    let pos =
+      coeffs[1..coded_tx_size].iter().rposition(|c| c.as_().abs() >= deadzone);
     // We skip the DC coefficient since it has its own quantizer index.
-    let last_pos = if is_zero { pos } else { pos + 1 };
+    let last_pos = pos.map(|pos| pos + 1).unwrap_or(1);
```

Comment:
What type does `pos` end up that this weird rigamarole is needed?

Reply:
Option